ratis-netty/src/main/java/org/apache/ratis/netty/server/NettyServerStreamRpc.java

```diff
@@ -199,25 +207,45 @@ public void channelRead(ChannelHandlerContext ctx, Object msg) throws IOExceptio
           // do not need to forward header request
           final List<DataStreamOutput> outs = proxies.getDataStreamOutput();
           peersStreamOutput.put(request.getStreamId(), outs);
-          for(DataStreamOutput out : outs) {
+          for (DataStreamOutput out : outs) {
             remoteWrites.add(out.getHeaderFuture());
           }
         } else {
           // body
-          for(DataStreamOutput out : peersStreamOutput.get(request.getStreamId())) {
+          for (DataStreamOutput out : peersStreamOutput.get(request.getStreamId())) {
             remoteWrites.add(out.writeAsync(request.slice().nioBuffer()));
           }
         }
 
         JavaUtils.allOf(remoteWrites).thenCombine(localWrite, (v, bytesWritten) -> {
               buf.release();
-              sendReply(request, bytesWritten, ctx);
```

Comment:
You can 

sendReply(request, bytesWritten, ctx);
 

 
You can move the following code into sendReply, and rename the origin sendReply to sendReplySuccess.
             ```
 try {
                if (!checkSuccessRemoteWrite(remoteWrites, bytesWritten)) {
                  sendReplyNotSuccess(request, ctx);
                } else {
                  sendReply(request, bytesWritten, ctx);
                }
              } catch (ExecutionException | InterruptedException e) {
                IOUtils.asIOException(e);
              }
```

Reply:
I see. Updated.